tests/src/test/java/org/eclipse/hono/tests/mqtt/MqttPublishTestBase.java

```diff
@@ -66,6 +67,13 @@
     // <MQTT message ID, PUBACK handler>
     private final Map<Integer, Handler<Integer>> pendingMessages = new HashMap<>();
 
+    /**
+     * Gets the QoS level with which the MQTT message shall be sent.
+     *
+     * @return The QoS level.
+     */
+    protected abstract MqttQoS getExpectedQos();
```

Comment:
then it is probably not the *expected* QoS but simply the QoS, right?

Reply:
Actually it is both as the QoS to be expected should be the one with the message was sent. I am fine with renaming though.